dace/optim/map_tiling_tuner.py

```diff
@@ -0,0 +1,97 @@
+# Copyright 2019-2022 ETH Zurich and the DaCe authors. All rights reserved.
+import copy
+
+import dace
+import math
+
+import itertools
+from dace.sdfg.sdfg import SDFG
+import numpy as np
+
+from dace.sdfg.analysis.cutout import cutout_state
+from dace.optim.cutout_tuner import CutoutTuner
+from dace.transformation.dataflow import MapTiling
+
+from dace.transformation.optimizer import Optimizer
+from dace.transformation.helpers import get_parent_map
+
+class MapTilingTuner(CutoutTuner):
+
+    def __init__(self, sdfg: dace.SDFG) -> None:
+        super().__init__(sdfg=sdfg)
+
+    def space(self, parent_map):
+        # TODO: choices
+        choices = [
+            (8, 8, 8),
+            (16, 16, 16),
+            (32, 32, 32),
+            (64, 64, 64),
+            (128, 128, 128),
+            (256, 256, 256),
+            (4, 8, 16),
+            (8, 16, 32),
+            (16, 32, 64),
+            (32, 64, 128),
+            (64, 128, 256),
+            (64, 8, 8)
+        ]
+
+        return choices
+
+    def optimize(self, measurements: int = 30):
+        dreport = self._sdfg.get_instrumented_data()
+        
+        tuning_report = {}
+        for state, parent_map in self.cutouts():
+            subgraph_nodes = state.scope_subgraph(parent_map).nodes()
+            cutout = cutout_state(state, *subgraph_nodes)
+            cutout.instrument = dace.InstrumentationType.Timer
+
+            arguments = {}
+            for cstate in cutout.nodes():
+                for dnode in cstate.data_nodes():
+                    if cutout.arrays[dnode.data].transient:
+                        continue
+
+                    arguments[dnode.data] = np.copy(dreport.get_first_version(dnode.data))
```

Comment:
why copy the data? it's not going to change, is it?

Reply:
TypeError: Passing a numpy view (e.g., sub-array or "A.T") to DaCe programs is not allowed in order to retain analyzability. Please make a copy with "numpy.copy(...)". If you know what you are doing, you can override this error in the configuration by setting compiler.allow_view_arguments to True.

Reply:
which is why I enable view arguments in the other tuning example. maybe we should enable it here too